test/python/unittest/API/APITest.py

```diff
@@ -413,6 +413,215 @@ def testConstantRecords(self):
+        )
+        self.assertEqual(
+            ms["INT"][SCALAR].shape,
+            [0 for _ in range(4)]
+        )
+        self.assertEqual(
+            ms["LONG"][SCALAR].shape,
+            [0 for _ in range(5)]
+        )
+        self.assertEqual(
+            ms["LONGLONG"][SCALAR].shape,
+            [0 for _ in range(6)]
+        )
+        self.assertEqual(
+            ms["USHORT"][SCALAR].shape,
+            [0 for _ in range(7)]
+        )
+        self.assertEqual(
+            ms["UINT"][SCALAR].shape,
+            [0 for _ in range(8)]
+        )
+        self.assertEqual(
+            ms["ULONG"][SCALAR].shape,
+            [0 for _ in range(9)]
+        )
+        self.assertEqual(
+            ms["ULONGLONG"][SCALAR].shape,
+            [0 for _ in range(10)]
+        )
+        self.assertEqual(
+            ms["FLOAT"][SCALAR].shape,
+            [0 for _ in range(11)]
+        )
+        self.assertEqual(
+            ms["DOUBLE"][SCALAR].shape,
+            [0 for _ in range(12)]
+        )
+        self.assertEqual(
+            ms["LONG_DOUBLE"][SCALAR].shape,
+            [0 for _ in range(13)]
+        )
+
+        if found_numpy:
+            self.assertEqual(
+                ms["int16"][SCALAR].shape,
+                [0 for _ in range(14)]
+            )
+            self.assertEqual(
+                ms["int32"][SCALAR].shape,
+                [0 for _ in range(15)]
+            )
+            self.assertEqual(
+                ms["int64"][SCALAR].shape,
+                [0 for _ in range(16)]
+            )
+            self.assertEqual(
+                ms["uint16"][SCALAR].shape,
+                [0 for _ in range(17)]
+            )
+            self.assertEqual(
+                ms["uint32"][SCALAR].shape,
+                [0 for _ in range(18)]
+            )
+            self.assertEqual(
+                ms["uint64"][SCALAR].shape,
+                [0 for _ in range(19)]
+            )
+            self.assertEqual(
+                ms["single"][SCALAR].shape,
+                [0 for _ in range(20)]
+            )
+            self.assertEqual(
+                ms["double"][SCALAR].shape,
+                [0 for _ in range(21)]
+            )
+
+        if file_ending == "bp":
+            return
+
+        self.assertTrue(
+            api.determine_datatype(ms["CHAR"][SCALAR].dtype) ==
+            DT.CHAR
+        )
+        self.assertTrue(
+            api.determine_datatype(ms["UCHAR"][SCALAR].dtype) ==
+            DT.UCHAR
+        )
+        self.assertTrue(
+            api.determine_datatype(ms["SHORT"][SCALAR].dtype) ==
+            DT.SHORT
+        )
+        self.assertTrue(
+            api.determine_datatype(ms["INT"][SCALAR].dtype) ==
+            DT.INT
+        )
+        self.assertTrue(
+            api.determine_datatype(ms["LONG"][SCALAR].dtype) ==
+            DT.LONG
+        )
+        self.assertTrue(
+            api.determine_datatype(ms["LONGLONG"][SCALAR].dtype) ==
```

Comment:
careful, this is not how `int`/`long`/`long long` works in C/C++: https://github.com/openPMD/openPMD-api/pull/333#issuecomment-419257519 / #342 

backends likely serialize them to the closest fixed size type, leading to a round-trip that depends on the [data model of a platform](https://en.cppreference.com/w/cpp/language/types#Integer_types).

Reply:
Thanks for the hint, I have changed things accordingly!